BuildingBlocks/test/BuildingBlocks.Application.Tests/Mediatr/QuotaEnforcerBehaviorTests.cs

```diff
@@ -2,202 +2,166 @@
 
         await Task.Run(acting, CancellationToken.None);
 
         // Assert
-        nextHasRan.Should().BeTrue();
+        nextMock.WasCalled.Should().BeTrue();
     }
-
+    
     [Fact]
-    public void Throws_QuotaExhaustedException_when_exactly_one_metric_is_exhausted()
+    public void Throws_QuotaExhaustedException_when_the_metric_is_still_exhausted()
     {
         // Arrange
-        var metricStatusesStubRepository = new MetricStatusesStubRepository(new() {
-            TestData.MetricStatus.ThatIsExhaustedFor1Day, TestData.MetricStatus.ThatWasExhaustedUntilYesterday
-        });
-
-        var behavior = CreateQuotaEnforcerBehavior(metricStatusesStubRepository);
+        var behavior = CreateQuotaEnforcerBehavior(TestData.MetricStatus.ThatIsExhaustedFor10Days);
+        var nextMock = new NextMock<TestData.IResponse>();
 
         // Act
         Func<Task> acting = async () => await behavior.Handle(
-            new TestCommand(),
-            () =>
-            {
-                return Task.FromResult(new IResponse());
-            },
+            new TestData.TestCommand(),
+            nextMock.Value,
             CancellationToken.None);
 
         // Assert
         acting.Should().AwaitThrowAsync<QuotaExhaustedException>()
-            .Which.ExhaustedMetricStatuses.First().MetricKey.Should().Be(TestData.MetricStatus.ThatIsExhaustedFor1Day.MetricKey);
+            .Which.ExhaustedMetricStatuses.First().MetricKey.Should().Be(TestData.MetricStatus.ThatIsExhaustedFor10Days.MetricKey);
     }
 
     [Fact]
-    public void Throws_QuotaExhaustedException_when_more_than_one_metric_is_exhausted()
+    public void Throws_QuotaExhaustedException_when_exactly_one_metric_is_exhausted()
     {
         // Arrange
-        var metricStatusesStubRepository = new MetricStatusesStubRepository(new() {
-            TestData.MetricStatus.ThatIsExhaustedFor1Day, TestData.MetricStatus.ThatIsExhaustedFor10Days
-        });
-
-        var behavior = CreateQuotaEnforcerBehavior(metricStatusesStubRepository);
+        var behavior = CreateQuotaEnforcerBehavior(
+            TestData.MetricStatus.ThatIsExhaustedFor1Day,
+            TestData.MetricStatus.ThatWasExhaustedUntilYesterday
+            );
+        var nextMock = new NextMock<TestData.IResponse>();
 
         // Act
         Func<Task> acting = async () => await behavior.Handle(
-            new TestCommand(),
-            () =>
-            {
-                return Task.FromResult(new IResponse());
-            },
+            new TestData.TestCommand(),
+            nextMock.Value,
             CancellationToken.None);
 
         // Assert
         acting.Should().AwaitThrowAsync<QuotaExhaustedException>()
-            .Which.ExhaustedMetricStatuses.All(it =>
-                it.IsExhaustedUntil > DateTime.Now
-            ).Should().BeTrue();
+            .Which.ExhaustedMetricStatuses.First().MetricKey.Should().Be(TestData.MetricStatus.ThatIsExhaustedFor1Day.MetricKey);
```

Comment:
You should probably assert that there is only one item in ExhaustedMetricStatuses. Check the length in the other tests as well